src/sentry/web/frontend/auth_organization_login.py

```diff
@@ -51,9 +52,15 @@ def handle(self, request: Request, organization_slug) -> Response:
         try:
             organization = Organization.objects.get(slug=organization_slug)
         except Organization.DoesNotExist:
-            return self.redirect(reverse("sentry-login"))
+            return self.handle_basic_auth(
+                request, organization=ApiOrganizationSummary(name=organization_slug)
+            )
```

Comment:
If we know that the organization doesn't exist, why redirect the user to an org specific login page?

Reply:
I've removed this.